src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -73,86 +73,156 @@ void KevinHallModel::generate_risk_factors(RuntimeContext &context) {
         initialise_weight(person);
     }
 
-    // Adjust weight to matche expected values.
+    // Adjust weight mean to match expected.
     adjust_risk_factors(context, {"Weight"_id});
 
+    // Compute weight power means by sex and age.
+    auto W_power_means = compute_mean_weight(context.population(), height_slope_);
+
     // Initialise everyone.
     for (auto &person : context.population()) {
+        double W_power_mean = W_power_means.at(person.gender, person.age);
+        initialise_height(person, W_power_mean, context.random());
         initialise_kevin_hall_state(person);
         compute_bmi(person);
     }
 }
 
 void KevinHallModel::update_risk_factors(RuntimeContext &context) {
 
-    // Initialise newborns and update others.
+    // Update (initialise) newborns.
+    update_newborns(context);
+
+    // Update non-newborns.
+    update_non_newborns(context);
+
+    // Compute BMI values for everyone.
     for (auto &person : context.population()) {
         // Ignore if inactive.
         if (!person.is_active()) {
             continue;
         }
 
-        if (person.age == 0) {
-            initialise_nutrient_intakes(person);
-            initialise_energy_intake(person);
-        } else {
-            update_nutrient_intakes(person);
-            update_energy_intake(person);
-        }
+        compute_bmi(person);
     }
+}
 
-    // Update: different model for children and adults.
+void KevinHallModel::update_newborns(RuntimeContext &context) const {
+
+    // Initialise nutrient and energy intake and weight for newborns.
     for (auto &person : context.population()) {
-        // Ignore if inactive.
+        // Ignore if inactive or not newborn.
         if (!person.is_active()) {
             continue;
         }
+        if (person.age != 0) {
+            continue;
+        }
```

Comment:
Just to visually reduce the amount of lines of code, yet keeping things simple:
```suggestion
        if (!person.is_active() || (person.age != 0)) {
            continue;
        }
```

Reply:
Ah, that was @alexdewar's suggestion. What's the tie breaker!?

Reply:
I tend to agree, though. All those `if`s are a bit verbose.

Reply:
I only suggested it because I thought it needed to be `if (person.age.has_value() && person.age != 0)`. If the whole thing fits on one line now, then that's def better!